packages/blockchain/src/index.ts

```diff
@@ -457,34 +622,46 @@ export default class Blockchain implements BlockchainInterface {
       }
 
       if (this._validateConsensus) {
-        if (this._ethash) {
-          const valid = await this._ethash.verifyPOW(block)
+        if (this._common.consensusAlgorithm() === 'ethash') {
+          const valid = await this._ethash!.verifyPOW(block)
           if (!valid) {
             throw new Error('invalid POW')
           }
         }
+        if (this._common.consensusAlgorithm() === 'clique') {
+          const valid = header.cliqueVerifySignature(this.cliqueActiveSigners())
+          if (!valid) {
+            throw new Error('invalid PoA block signature (clique)')
+          }
+        }
       }
 
-      // set total difficulty in the current context scope
-      if (this._headHeaderHash) {
-        currentTd.header = await this.getTotalDifficulty(this._headHeaderHash)
-      }
-      if (this._headBlockHash) {
-        currentTd.block = await this.getTotalDifficulty(this._headBlockHash)
-      }
+      if (this._common.consensusAlgorithm() === 'ethash') {
+        // set total difficulty in the current context scope
+        if (this._headHeaderHash) {
+          currentTd.header = await this.getTotalDifficulty(this._headHeaderHash)
+        }
+        if (this._headBlockHash) {
+          currentTd.block = await this.getTotalDifficulty(this._headBlockHash)
+        }
 
-      // calculate the total difficulty of the new block
-      const parentTd = await this.getTotalDifficulty(header.parentHash, blockNumber.subn(1))
-      td.iadd(parentTd)
+        // calculate the total difficulty of the new block
+        const parentTd = await this.getTotalDifficulty(header.parentHash, blockNumber.subn(1))
+        td.iadd(parentTd)
+      }
 
-      // save block and total difficulty to the database
+      // save total difficulty to the database
       dbOps = dbOps.concat(DBSetTD(td, blockNumber, blockHash))
```

Comment:
So should this always be executed or could this also be put into the ethash-only section? 🤔

Reply:
probably can be moved inside, I wasn't sure if it would have a problem retrieving it later with a key NotFoundError, but I will check.

Reply:
added a commit for this (https://github.com/ethereumjs/ethereumjs-monorepo/pull/1032/commits/805a97361d7a174167160a3375b0a1cc0eff975c)